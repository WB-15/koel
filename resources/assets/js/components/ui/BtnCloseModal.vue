<template>
  <button data-testid="close-modal-btn" @click.prevent="$emit('click')">
    <i class="fa fa-times"></i>
  </button>
</template>

<style lang="scss" scoped>
button {
  display: flex;
  place-content: center;
  place-items: center;
  position: absolute;
  right: 0;
  top: 0;
  background: var(--color-red);
  width: 24px;
  height: 24px;
  border-radius: 0;
  padding: 0;
  color: var(--color-text-primary);
  transition: opacity .3s;

  @media (hover: none) {
    opacity: 1;
  }
}
</style>
