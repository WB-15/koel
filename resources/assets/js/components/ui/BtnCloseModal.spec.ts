import { expect, it } from 'vitest'
import { fireEvent } from '@testing-library/vue'
import UnitTestCase from '@/__tests__/UnitTestCase'
import BtnCloseModal from './BtnCloseModal.vue'

new class extends UnitTestCase {
  protected test () {
    it('renders', () => expect(this.render(BtnCloseModal).html()).toMatchSnapshot())

    it('emits the event', async () => {
      const { emitted, getByRole } = this.render(BtnCloseModal)

      await fireEvent.click(getByRole('button'))

      expect(emitted().click).toBeTruthy()
    })
  }
}
