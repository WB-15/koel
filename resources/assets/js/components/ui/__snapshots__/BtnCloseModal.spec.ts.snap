// Vitest Snapshot v1

exports[`renders 1`] = `<button data-testid="close-modal-btn" type="button" data-v-8e854c66=""><i class="fa fa-times" data-v-8e854c66=""></i></button>`;
